psl_class.py

```diff
@@ -142,18 +126,24 @@ def bind(self, interface):
             return False
         self.srcmac = pack_mac(get_hw_addr(interface))
 
-        # send socket
-        self.ssocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        # send socket, can also be used to receive unicast!
+        self.ssocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
         self.ssocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.ssocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
         self.ssocket.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+
         self.ssocket.bind((self.myhost, self.RECPORT))
+        #self.ssocket.bind(('<broadcast>', self.RECPORT))
+        #self.ssocket.bind(('', self.RECPORT))
 
-        # receive socket
+        # separate broadcast receive socket
         self.rsocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self.rsocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        # self.rsocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        self.rsocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
         self.rsocket.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-        self.rsocket.bind(("255.255.255.255", self.RECPORT))
+        #self.rsocket.bind(("<broadcast>", self.RECPORT))
```

Comment:
Left-over testing code?

Reply:
Hm... from the Python documentation:
> For IPv4 addresses, two special forms are accepted instead of a host address: `''` represents INADDR_ANY, which is used to bind to all interfaces, and the string `'<broadcast>'` represents INADDR_BROADCAST. This behavior is not compatible with IPv6, therefore, you may want to avoid these if you intend to support IPv6 with your Python programs.

https://docs.python.org/3/library/socket.html

Looking at the Python source code these are treated the same:

https://github.com/python/cpython/blob/8b55adfa8ff05477b4be7def36db7b66c73f181d/Modules/socketmodule.c#L1166-L1175

So we should delete one of these (I think `"<broadcast>"` is nicer to read, as it shows the intention to coders who may not be familiar with the mysteries of IPv4 special-case numbers), and remove the comment in any case.